### DISCUSSION TITLE
how does logging work?

### DISCUSSION THREAD

**Discussion by roncking** [langchain4j/langchain4j#177] [General]
Hi, I have included langchain4j in my maven based project. It runs fine, but I can't see any logging output at all! Here's my code:

        ChatLanguageModel chatLanguageModel = OpenAiChatModel.builder()
                .apiKey(ApiKeys.OPENAI_API_KEY)
                .modelName("gpt-4-0613") //gpt-4-0163
                .maxTokens(7000)
                .topP(1.0)
                .temperature(1.0)
                .timeout(ofSeconds(120))
                .logRequests(true)
                .logResponses(true)
                .build();

What do I need to do to activate logging, and where will the output be?

**Comment by benzourry** [comment]
I'm using normal java version of langchain4j with spring boot (without using langchain4j spring boot starter) with default spring boot logging.

As mention by @langchain4j, i set

```
.logRequests(true)
.logResponses(true)
```

and in my application.properties, I just add
```
logging.level.dev.langchain4j=DEBUG
```

**Reply by jscksy** [reply]
> I'm using normal java version of langchain4j with spring boot (without using langchain4j spring boot starter) with default spring boot logging.
> 
> As mention by @langchain4j, i set
> 
> ```
> .logRequests(true)
> .logResponses(true)
> ```
> 
> and in my application.properties, I just add
> 
> ```
> logging.level.dev.langchain4j=DEBUG
> ```

hello，did that work？i can't print the logs too.

**Reply by jscksy** [reply]
> @jscksy do you see any other logs (not from LangChain4j)?
yes, like below
[main] INFO
FortuneAl.BootStrapApplication - No
active profile set, falling back to 1
default profile: "default"
Imain] INFO
org.springframework.boot.web.embe
dded.tomcat.TomcatWebServer-
Tomcat initialized with port(s):8080
(http)
[main] INFO
org.apache.catalina.core.StandardSe
rvice - Starting service [Tomcat]
Imain] INFO
org.apache.catalina.core.StandardEn
gine - Starting Servlet engine:
[Apache Tomcat/9.0.69]
[main] INFO
org.apache.catalina.core.ContainerBa
se.[Tomcat].[localhost].[/]-
Initializing Spring embedded
WebApplicationContext
[main] INFO
org.springframework.boot.web.servle t.context.ServletWebServerApplicati onContext -Root
WebApplicationContext: initialization

**Reply by jscksy** [reply]
> @jscksy do you see any other logs (not from LangChain4j)?

i had done all the settings above

**Reply by jscksy** [reply]
https://github.com/jscksy/LLMApplication 
a simple demo
through the class FortuneAiAssistant to access LLM

**Reply by jscksy** [reply]
thank u at fisrt ,i use the controller to call the method,and through the idea console to input information.It might not be a good way.

I add a "ChatTest" Class in the test folder,just simply use a loop to get input and query the LLM
